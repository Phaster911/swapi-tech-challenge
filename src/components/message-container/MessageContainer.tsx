import ErrorMessage from "./messages/error-message/ErrorMessage";
import NoResultsMessage from "./messages/no-results-message/NoResultsMessage";
import LoadingMessage from "./messages/loading-message/LoadingMessage";
import StyledMessageWrapper from "../shared/styled-message-wrapper/StyledMessageWrapper";

export type MessageContainerProps = {
  isPending: boolean;
  isError: boolean;
  isSuccess: boolean;
  pageCount: number;
  tryAgainHandler: () => void;
};

export default function MessageContainer({
  isPending,
  isError,
  isSuccess,
  pageCount,
  tryAgainHandler,
}: MessageContainerProps) {
  return (
    <section>
      <StyledMessageWrapper>
        {isPending && <LoadingMessage />}
        {isError && <ErrorMessage tryAgainHandler={tryAgainHandler} />}
        {isSuccess && pageCount === 0 && <NoResultsMessage />}
      </StyledMessageWrapper>
    </section>
  );
}
