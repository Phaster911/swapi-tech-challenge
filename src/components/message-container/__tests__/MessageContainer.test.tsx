import { render } from "@testing-library/react";
import MessageContainer, { MessageContainerProps } from "../MessageContainer";

vi.mock("../messages/no-results-message/NoResultsMessage", () => {
  return {
    default: () => <div>NoResultsMessage</div>,
  };
});

vi.mock("../messages/error-message/ErrorMessage", () => {
  return {
    default: () => <div>ErrorMessage</div>,
  };
});

vi.mock("../messages/loading-message/LoadingMessage", () => {
  return {
    default: () => <div>LoadingMessage</div>,
  };
});

describe("<MessageContainer />", () => {
  let props: MessageContainerProps;
  let tryAgainFn: ReturnType<typeof vi.fn>;

  beforeEach(() => {
    tryAgainFn = vi.fn();
  });

  describe("when isPending is true", () => {
    beforeEach(() => {
      props = {
        isPending: true,
        isError: false,
        isSuccess: false,
        pageCount: 0,
        tryAgainHandler: tryAgainFn,
      };
    });

    it("renders the loading message", () => {
      const { container } = render(<MessageContainer {...props} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe("when isError is true", () => {
    beforeEach(() => {
      props = {
        isPending: false,
        isError: true,
        isSuccess: false,
        pageCount: 0,
        tryAgainHandler: tryAgainFn,
      };
    });
    it("renders the error message", () => {
      const { container } = render(<MessageContainer {...props} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe("when isSuccess is true and there are no results", () => {
    beforeEach(() => {
      props = {
        isPending: false,
        isError: false,
        isSuccess: true,
        pageCount: 0,
        tryAgainHandler: tryAgainFn,
      };
    });
    it("renders the no results message", () => {
      const { container } = render(<MessageContainer {...props} />);

      expect(container).toMatchSnapshot();
    });
  });
});
