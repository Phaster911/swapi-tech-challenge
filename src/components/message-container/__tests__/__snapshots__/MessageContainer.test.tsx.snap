// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MessageContainer /> > when isError is true > renders the error message 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1rem;
  text-align: center;
}

<div>
  <section>
    <div
      class="c0"
    >
      <div>
        ErrorMessage
      </div>
    </div>
  </section>
</div>
`;

exports[`<MessageContainer /> > when isPending is true > renders the loading message 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1rem;
  text-align: center;
}

<div>
  <section>
    <div
      class="c0"
    >
      <div>
        LoadingMessage
      </div>
    </div>
  </section>
</div>
`;

exports[`<MessageContainer /> > when isSuccess is true and there are no results > renders the no results message 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1rem;
  text-align: center;
}

<div>
  <section>
    <div
      class="c0"
    >
      <div>
        NoResultsMessage
      </div>
    </div>
  </section>
</div>
`;
